src/helpers/locationProps.ts

```diff
@@ -0,0 +1,72 @@
+  export interface LocationProps {
```

Comment:
This may need to be used where ever all our data is coming in, it is currently set to except undefined just so I could get past TS errors, we may want to rename so its a more universal helper.

Reply:
This is awesome and so organized, great work!